ui.frontend/src/view/FormFileInputWidgetBase.js

```diff
@@ -177,9 +175,23 @@ class FormFileInputWidgetBase {
                     // todo: add support here
                     //let previewFileObjIdx = this._getFileObjIdx(index);
                     let previewFile = this.fileArr[index]?.data;
-                    let objectUrl = FileInputWidget.previewFileUsingObjectUrl(previewFile);
-                    if (objectUrl) {
-                        elem.dataset.objectUrl = objectUrl;
+                    if (this.fileArr[index] && this.fileArr[index].type === 'image/svg+xml') {
+                        const reader = new FileReader();
+                        reader.onload = function(e) {
```

Comment:
We don't read file attachment inline in browser, this would impact peformance, Please use the object URL directly and use the approach suggested in the issue description.